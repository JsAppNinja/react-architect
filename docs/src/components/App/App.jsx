import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { NavigationDrawer, bem } from 'react-md';

import navItems from 'constants/navItems';
import Link from 'components/Link';
import DocumentationTabs from 'components/DocumentationTabs';
import Messages from 'components/Messages';
import Search from 'components/Search';
import MobileNavigation from 'components/MobileNavigation';

import History from './History';
import Routes from './Routes';
import Footer from './Footer';

const helmetConfig = {
  htmlAttributes: { lang: 'en', class: 'custom-theme' },
  defaultTitle: 'react-md',
  titleTemplate: '%s - react-md',
};

const App = ({
  mobile,
  mobileNavigation,
  defaultMedia,
  toolbarTitle,
  toolbarProminent,
  visibleBoxShadow,
  contentProps,
  meta,
  link,
  searching,
  updateMedia,
  animationKey,
}) => {
  let bottomNav;
  if (mobileNavigation) {
    bottomNav = <MobileNavigation key="bottom-navigation" />;
  }

  return (
    <NavigationDrawer
      drawerId="main-navigation"
      drawerTitle="react-md"
      defaultMedia={defaultMedia}
      onMediaTypeChange={updateMedia}
      toolbarId="main-toolbar"
      toolbarTitle={toolbarTitle}
      toolbarTitleClassName={bem('main-toolbar', 'title', { 'minified': searching })}
      toolbarZDepth={visibleBoxShadow ? undefined : 0}
      toolbarProminent={!mobile && toolbarProminent}
      toolbarChildren={<DocumentationTabs visible={!mobile && toolbarProminent} />}
      toolbarActions={<Search key="search" />}
      contentProps={contentProps}
      navItems={navItems.map(({ divider, subheader, ...route }) => {
        if (divider || subheader) {
          return { divider, subheader, ...route };
        }

        return <Link {...route} />;
      })}
    >
      <Helmet {...helmetConfig} title={toolbarTitle} meta={meta} link={link} />
      <History />
      <Routes key={animationKey} />
      {bottomNav}
      <Messages />
      <Footer />
    </NavigationDrawer>
  );
};

App.propTypes = {
  animationKey: PropTypes.string.isRequired,
  mobile: PropTypes.bool.isRequired,
  mobileNavigation: PropTypes.bool.isRequired,
  defaultMedia: PropTypes.string.isRequired,
  toolbarTitle: PropTypes.string.isRequired,
  toolbarProminent: PropTypes.bool.isRequired,
  visibleBoxShadow: PropTypes.bool.isRequired,
  contentProps: PropTypes.object,
  meta: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
  })).isRequired,
  link: PropTypes.arrayOf(PropTypes.shape({
    rel: PropTypes.string.isRequired,
    href: PropTypes.string.isRequired,
  })).isRequired,
  searching: PropTypes.bool.isRequired,
  updateMedia: PropTypes.func.isRequired,
};
export default App;
