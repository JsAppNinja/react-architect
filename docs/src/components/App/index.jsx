/**
 * There is probably a real way to do this, but separating the files into
 * App, History, and ConnectedApp were the only way I could get hot reloading
 * to work.
 */
import ConnectedApp from './ConnectedApp';

export default ConnectedApp;
