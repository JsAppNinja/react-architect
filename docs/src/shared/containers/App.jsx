import React, { PureComponent, PropTypes, cloneElement } from 'react';
import { findDOMNode } from 'react-dom';
import cn from 'classnames';
import { connect } from 'react-redux';
import NavigationDrawer from 'react-md/lib/NavigationDrawers';

import { updateMedia, setCustomTheme } from 'actions/ui';
import { THEME_STORAGE_KEY } from 'constants/application';
import hasStorage from 'utils/hasStorage';
import loadCustomTheme from 'utils/loadCustomTheme';
import navItems from 'constants/navItems';
import DocumentationTabs from 'containers/DocumentationTabs';
import Notifications from 'containers/Notifications';
import AppFooter from 'containers/AppFooter';
import Search from 'containers/Search';

@connect(({ ui: { drawer } }) => ({ ...drawer }), {
  onMediaTypeChange: updateMedia,
  setCustomTheme,
})
export default class App extends PureComponent {
  static propTypes = {
    // Injected from connect and drawer state
    defaultMedia: NavigationDrawer.propTypes.defaultMedia,
    visibleBoxShadow: PropTypes.bool.isRequired,
    visibleToolbarTitle: PropTypes.bool.isRequired,
    toolbarTitle: PropTypes.string.isRequired,
    toolbarProminent: PropTypes.bool.isRequired,
    onMediaTypeChange: PropTypes.func.isRequired,
    setCustomTheme: PropTypes.func.isRequired,
    customDrawerType: PropTypes.string,

    params: PropTypes.shape({
      component: PropTypes.string,
      section: PropTypes.string,
    }).isRequired,
    location: PropTypes.shape({
      pathname: PropTypes.string.isRequired,
    }).isRequired,
    className: PropTypes.string,
    children: PropTypes.node,
  };

  constructor(props) {
    super(props);

    this.state = {
      minOffset: 256 + 64,
    };

    this._setContainer = this._setContainer.bind(this);
    this._calcMinHeight = this._calcMinHeight.bind(this);
    this._handleMediaTypeChange = this._handleMediaTypeChange.bind(this);
  }

  componentWillMount() {
    if (hasStorage() && localStorage.getItem(THEME_STORAGE_KEY) !== null) {
      this.props.setCustomTheme(true);
    }
  }

  componentDidMount() {
    if (hasStorage() && localStorage.getItem(THEME_STORAGE_KEY) !== null) {
      const { primary, secondary, light, hue } = JSON.parse(localStorage.getItem(THEME_STORAGE_KEY));
      loadCustomTheme(primary, secondary, hue, light);
    }
  }

  componentDidUpdate(prevProps) {
    if (this.props.toolbarProminent !== prevProps.toolbarProminent) {
      this._calcMinHeight();
    }
  }

  _setContainer(container) {
    this._container = findDOMNode(container);
    this._calcMinHeight();
  }

  _calcMinHeight() {
    if (!this._container) {
      return;
    }

    const { offsetHeight: toolbarHeight } = this._container.querySelector('.main-toolbar');
    const { offsetHeight: footerHeight } = this._container.querySelector('.app-footer');
    this.setState({ minOffset: toolbarHeight + footerHeight });
  }

  _handleMediaTypeChange(drawerType, media) {
    this._calcMinHeight();
    this.props.onMediaTypeChange(drawerType, media);
  }

  render() {
    const {
      params,
      location: { pathname },
      defaultMedia,
      visibleBoxShadow,
      visibleToolbarTitle,
      toolbarTitle,
      toolbarProminent,
      customDrawerType,
    } = this.props;

    const { minOffset } = this.state;

    let { children } = this.props;
    if (children) {
      children = cloneElement(children, { key: pathname, style: { minHeight: `calc(100vh - ${minOffset}px)` } });
    }

    let tabs;
    if (toolbarProminent) {
      tabs = <DocumentationTabs key="tabs" params={params} pathname={pathname} />;
    }

    return (
      <NavigationDrawer
        ref={this._setContainer}
        drawerTitle="react-md"
        defaultMedia={defaultMedia}
        toolbarClassName={cn('main-toolbar', { 'main-toolbar--title-hidden': !visibleToolbarTitle })}
        toolbarTitle={toolbarTitle}
        toolbarProminent={toolbarProminent}
        toolbarChildren={tabs}
        navItems={navItems(pathname)}
        drawerType={customDrawerType}
        toolbarStyle={!visibleBoxShadow ? { boxShadow: 'none' } : null}
        toolbarActions={<Search key="search" />}
        onMediaTypeChange={this._handleMediaTypeChange}
      >
        {children}
        <AppFooter key="footer" home={pathname === '/'} />
        <Notifications key="notifications" />
      </NavigationDrawer>
    );
  }
}
