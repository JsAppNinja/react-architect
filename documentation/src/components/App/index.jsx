import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connectAdvanced } from 'react-redux';
import { withRouter } from 'react-router';
import shallowEqual from 'shallowequal';
import Helmet from 'react-helmet';
import NavigationDrawer from 'react-md/lib/NavigationDrawers';

import { updateMedia } from 'state/media';
import { updateLocation } from 'state/routing';
import navItems from 'constants/navItems';
import scrollRestoration from 'utils/scrollRestoration';
import Link from 'components/Link';
import DocumentationTabs from 'components/DocumentationTabs';
import Search from 'components/Search';

import Routes from './Routes';
import Footer from './Footer';

const helmetConfig = {
  htmlAttributes: { lang: 'en', class: 'custom-theme' },
  defaultTitle: 'react-md',
  titleTemplate: '%s - react-md',
};

@withRouter
@connectAdvanced((dispatch) => {
  let result;

  return (state, props) => {
    const {
      drawer,
      helmet,
      media: { defaultMedia, mobile },
    } = state;

    const nextResult = {
      ...props,
      ...drawer,
      ...helmet,
      dispatch,
      defaultMedia,
      mobile,
    };

    if (!shallowEqual(result, nextResult)) {
      result = nextResult;
    }

    return result;
  };
})
export default class App extends PureComponent {
  static propTypes = {
    mobile: PropTypes.bool.isRequired,
    defaultMedia: PropTypes.string.isRequired,
    toolbarTitle: PropTypes.string.isRequired,
    toolbarProminent: PropTypes.bool.isRequired,
    visibleBoxShadow: PropTypes.bool.isRequired,
    location: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
    children: PropTypes.node,
    meta: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      content: PropTypes.string.isRequired,
    })).isRequired,
    link: PropTypes.arrayOf(PropTypes.shape({
      rel: PropTypes.string.isRequired,
      href: PropTypes.string.isRequired,
    })).isRequired,

    dispatch: PropTypes.func.isRequired,
  };

  componentDidMount() {
    const { history, dispatch } = this.props;
    history.listen((location) => {
      dispatch(updateLocation(location));
      if (typeof window.ga !== 'undefined') {
        window.ga('send', 'pageview', location.pathname);
      }
    });
    scrollRestoration();
  }

  componentDidUpdate() {
    scrollRestoration();
  }

  updateMedia = (drawerType, media) => {
    this.props.dispatch(updateMedia(drawerType, media));
  };

  render() {
    const {
      defaultMedia,
      toolbarTitle,
      visibleBoxShadow,
      toolbarProminent,
      mobile,
      meta,
      link,
    } = this.props;

    let tabs;
    if (!mobile && toolbarProminent) {
      tabs = <DocumentationTabs key="tabs" />;
    }

    return (
      <NavigationDrawer
        drawerId="main-navigation"
        drawerTitle="react-md"
        toolbarTitle={toolbarTitle}
        defaultMedia={defaultMedia}
        onMediaTypeChange={this.updateMedia}
        toolbarZDepth={visibleBoxShadow ? undefined : 0}
        toolbarProminent={!mobile && toolbarProminent}
        toolbarChildren={tabs}
        toolbarActions={<Search key="search" />}
        navItems={navItems.map(({ divider, subheader, ...route }) => {
          if (divider || subheader) {
            return { divider, subheader, ...route };
          }

          return <Link {...route} />;
        })}
      >
        <Helmet {...helmetConfig} title={toolbarTitle} meta={meta} link={link} />
        <Routes />
        <Footer />
      </NavigationDrawer>
    );
  }
}
