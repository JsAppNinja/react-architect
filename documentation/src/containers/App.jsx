import React, { PureComponent, PropTypes } from 'react';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';
import NavigationDrawer from 'react-md/lib/NavigationDrawers';

import { updateMedia } from 'state/media';

const helmetConfig = {
  htmlAttributes: { lang: 'en' },
  defaultTitle: 'react-md',
  titleTemplate: '%s - react-md',
  meta: [{
    name: 'description',
    content: 'Google\'s Material Design UI Components built with accessibility in mind, Sass, And React.',
  }, {
    name: 'keywords',
    content: 'react-md,material design,react,components,material ui',
  }],
  link: [{
    rel: 'stylesheet',
    href: 'https://fonts.googleapis.com/css?family=Roboto:400,500,700|Material+Icons',
  }, {
    rel: 'stylesheet',
    href: 'https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.min.css',
  }],
};

@connect(({ media: { defaultMedia } }) => ({ defaultMedia }))
export default class App extends PureComponent {
  static propTypes = {
    defaultMedia: PropTypes.string.isRequired,

    dispatch: PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);

    this.state = {};
  }

  updateMedia = (drawerType, media) => {
    this.props.dispatch(updateMedia(drawerType, media));
  };

  render() {
    const { defaultMedia } = this.props;
    return (
      <NavigationDrawer
        drawerTitle="react-md"
        toolbarTitle="Woop"
        defaultMedia={defaultMedia}
        onMediaTypeChange={this.updateMedia}
      >
        <Helmet {...helmetConfig} />
        <h1>Hello, world!</h1>
      </NavigationDrawer>
    );
  }
}
