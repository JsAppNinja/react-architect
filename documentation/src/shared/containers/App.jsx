import React, { PureComponent, PropTypes, cloneElement } from 'react';
import { connect } from 'react-redux';
import NavigationDrawer from 'react-md/lib/NavigationDrawers';

import navItems from 'constants/navItems';
import DocumentationTabs from 'containers/DocumentationTabs';
import Notifications from 'containers/Notifications';
import AppFooter from 'containers/AppFooter';
import { updateMedia } from 'actions/ui';

@connect(({ ui: { drawer } }) => ({ ...drawer }), {
  onMediaTypeChange: updateMedia,
})
export default class App extends PureComponent {
  static propTypes = {
    // Injected from connect and drawer state
    defaultMedia: NavigationDrawer.propTypes.defaultMedia,
    visibleBoxShadow: PropTypes.bool.isRequired,
    toolbarTitle: PropTypes.string.isRequired,
    toolbarProminent: PropTypes.bool.isRequired,
    onMediaTypeChange: PropTypes.func.isRequired,

    params: PropTypes.shape({
      component: PropTypes.string,
      section: PropTypes.string,
    }).isRequired,
    location: PropTypes.shape({
      pathname: PropTypes.string.isRequired,
    }).isRequired,
    className: PropTypes.string,
    children: PropTypes.node,
  };

  render() {
    const {
      params,
      location: { pathname },
      defaultMedia,
      visibleBoxShadow,
      toolbarTitle,
      toolbarProminent,
      onMediaTypeChange,
    } = this.props;

    let { children } = this.props;
    if (children) {
      children = cloneElement(children, { key: pathname });
    }

    let tabs;
    if (toolbarProminent) {
      tabs = <DocumentationTabs params={params} pathname={pathname} />;
    }

    return (
      <NavigationDrawer
        drawerTitle="react-md"
        defaultMedia={defaultMedia}
        toolbarTitle={toolbarTitle}
        toolbarTitleClassName="md-text-capitalize"
        toolbarProminent={toolbarProminent}
        toolbarChildren={tabs}
        navItems={navItems(pathname)}
        toolbarStyle={!visibleBoxShadow ? { boxShadow: 'none' } : null}
        onMediaTypeChange={onMediaTypeChange}
      >
        {children}
        <AppFooter key="footer" home={pathname === '/'} />
        <Notifications key="notifications" />
      </NavigationDrawer>
    );
  }
}
