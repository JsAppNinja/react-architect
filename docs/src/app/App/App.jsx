import React, { Component, PropTypes } from 'react';
import marked from 'marked';
import classnames from 'classnames';
import { Link } from 'react-router';
import Toolbar, { ActionArea } from 'react-md/lib/Toolbars';
import Avatar from 'react-md/lib/Avatars';
import FontIcon from 'react-md/lib/FontIcons';
import { IconButton } from 'react-md/lib/Buttons';
import Sidebar from 'react-md/lib/Sidebars';
import { List, ListItem, ListSubheader } from 'react-md/lib/Lists';
import Divider from 'react-md/lib/Dividers';
import { isMobile } from 'react-md/lib/utils';

import './_app.scss';
import '../Documentation/_markdown.scss';
import '../Documentation/_prop-types.scss';
import '../Documentation/_documentation.scss';

import * as components from '../components';
import { githubHref, hostPrefix, imgPrefix } from '../utils';
import GettingStarted from '../GettingStarted';
import Customization from '../Customization';
import Typography from '../Typography';

const componentLinks = Object.keys(components).map(k => {
  if(!components[k] || !components[k].name) { return; }

  const name = k.split(/(?=[A-Z])/);
  return {
    link: 'components/' + name.map(n => n.toLowerCase()).join('-'),
    label: name.join(' '),
  };
}).filter(l => !!l);

export default class App extends Component {
  constructor(props) {
    super(props);

    // Not home and not a media device
    const isOpen = props.location.pathname !== '/' && !isMobile;
    this.state = { isOpen };
  }

  static propTypes = {
    children: PropTypes.node,
    location: PropTypes.object, // from react-router
  };

  componentWillMount() {
    marked.setOptions({
      renderer: new marked.Renderer(),
      gfm: true,
      tables: true,
      breaks: false,
      pedantic: false,
      sanitize: false,
      smartLists: true,
      smartypants: false,
      highlight: (code, lang) => require('highlight.js').highlight(lang, code).value, // eslint-disable-line no-undef
    });
  }

  componentWillUpdate({ location }, { isOpen }) {
    const { pathname } = this.props.location;
    if(pathname === location.pathname) { return; }

    const isRoot = location.pathname === '/';
    if(!isRoot && !isOpen && !isMobile) {
      this.setState({ isOpen: true });
    } else if(isRoot && isOpen || isMobile) {
      setTimeout(() => {
        this.setState({ isOpen: false });
      }, 150);
    }
  }

  toggleMenu = () => {
    this.setState({ isOpen: !this.state.isOpen });
  };

  listItemClassName = (path) => {
    return classnames({ 'active': this.props.location.pathname === `/${path}` });
  };

  render() {
    const pathname = this.props.location.pathname;
    let pageTitle;
    switch(pathname) {
      case `/${GettingStarted.path}`:
        pageTitle = 'Getting Started';
        break;
      case `/${Customization.path}`:
        pageTitle = 'Customization';
        break;
      case `/${Typography.path}`:
        pageTitle = 'Typography';
        break;
      case '/':
        pageTitle = 'react-md';
        break;
      default:
        pageTitle = 'Components';
    }

    return (
      <div className="react-md-docs">
        <Toolbar
          primary
          fixed
          title={pageTitle}
          className="react-md-docs-toolbar"
          actionLeft={<IconButton onClick={this.toggleMenu}>menu</IconButton>}
          actionsRight={(
            <ActionArea>
              <IconButton href={githubHref} iconClassName="fa fa-github" />
            </ActionArea>
          )}
        />
        <Sidebar isOpen={this.state.isOpen} className="main-sidebar">
          <List>
            <ListItem
              component={Link}
              className={this.listItemClassName('')}
              to="/"
              primaryText="Home"
              key="home-link"
              leftIcon={<FontIcon>home</FontIcon>}
            />
            <ListItem
              component={Link}
              className={this.listItemClassName(GettingStarted.path)}
              to={`/${GettingStarted.path}`}
              primaryText="Getting Started"
              key="getting-started"
              leftIcon={<FontIcon>info_outline</FontIcon>}
            />
            <ListItem
              component={Link}
              className={this.listItemClassName(Customization.path)}
              to={`/${Customization.path}`}
              primaryText="Customization"
              key="customization"
              leftIcon={<FontIcon>build</FontIcon>}
            />
            <ListItem
              component={Link}
              className={this.listItemClassName(Typography.path)}
              to={`/${Typography.path}`}
              primaryText="Typography"
              key="typography"
              leftIcon={<FontIcon>text_fields</FontIcon>}
            />
            <Divider />
            <ListSubheader primaryText="References" />
            <ListItem
              component="a"
              primaryText="SASS Doc"
              href={`${hostPrefix}/sassdoc`}
              key="sassdoc"
              leftAvatar={<Avatar src={`${imgPrefix}/sass-icon.png`} alt="SASS Icon" />}
            />
            <ListItem
              component="a"
              primaryText="React"
              href="https://facebook.github.io/react/"
              key="react"
              leftAvatar={<Avatar src="https://facebook.github.io/react/img/logo.svg" alt="React logo" />}
            />
            <ListItem
              component="a"
              primaryText="Material Design"
              href="https://www.google.com/design/spec/material-design/introduction.html"
              key="material-design"
              leftAvatar={<Avatar src="https://i.ytimg.com/vi/PAKCgvprpQ8/maxresdefault.jpg" alt="Google logo" />}
            />
            <Divider />
            <ListSubheader primaryText="Components" />
          </List>
          <List>
            {componentLinks.map(({ link, label }) => {
              return (
              <ListItem
                component={Link}
                to={`/${link}`}
                className={classnames({ 'active': `/${link}` === pathname })}
                key={link}
                primaryText={label}
              />
              );
            })}
          </List>
        </Sidebar>
        <main className={classnames({ 'active': this.state.isOpen })}>
          {React.cloneElement(this.props.children, { key: pathname, marked })}
        </main>
      </div>
    );
  }
}
