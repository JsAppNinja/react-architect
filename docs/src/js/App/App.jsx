import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';
import classnames from 'classnames';

import { githubHref, mainLinks } from '../utils';
import * as components from '../components';
import { AppBar, IconButton, Sidebar, List, ListItem, ListDivider, ListSubheader } from 'react-md';

import './_app.scss';

const componentLinks = Object.keys(components).map(k => {
  if(!components[k]) { return; }

  const name = k.split(/(?=[A-Z])/);
  return {
    link: name.map(n => n.toLowerCase()).join('-'),
    label: name.join(' '),
  };
}).filter(l => !!l);

export default class App extends Component {
  constructor(props) {
    super(props);

    // Not home and not a media device
    const isOpen = props.location.pathname !== '/' && !App.isMobile();
    this.state = { isOpen };
  }

  static propTypes = {
    children: PropTypes.node,
    location: PropTypes.object, // from react-router
  };

  static isMobile() {
    return window.matchMedia('(max-width: 600px)').matches;
  }

  componentWillUpdate({ location }, { isOpen }) {
    const { pathname } = this.props.location;
    if(pathname === location.pathname) { return; }

    const isRoot = location.pathname === '/';
    if(!isRoot && !isOpen && !App.isMobile()) {
      this.setState({ isOpen: true });
    } else if(isRoot && isOpen || App.isMobile()) {
      this.setState({ isOpen: false });
    }
  }

  toggleMenu = () => {
    this.setState({ isOpen: !this.state.isOpen });
  };

  render() {
    const pathname = this.props.location.pathname;
    return (
      <div className="react-md-docs">
        <AppBar
          title="react md"
          className="react-md-docs-app-bar"
          leftNode={<IconButton onClick={this.toggleMenu}>menu</IconButton>}
          rightNode={<IconButton href={githubHref} iconClassName="fa fa-github" />}
        />
        <Sidebar isOpen={this.state.isOpen} className="main-sidebar">
          <List>
            {mainLinks.map(({ link, ...props }) => (
              <ListItem
                component={Link}
                className={`/${link}` === pathname ? 'active' : null}
                to={`/${link}`}
                key={link || 'home-link'}
                {...props}
              />
            ))}
            <ListDivider />
            <ListSubheader primaryText="Components" />
          </List>
          <List className="scrollable">
            {componentLinks.map(({ link, label }) => {
              return (
              <ListItem
                component={Link}
                to={`/${link}`}
                className={classnames('md-list-tile', {
                  'active': `/${link}` === pathname,
                })}
                key={link}
                primaryText={label}
              />
              );
            })}
          </List>
        </Sidebar>
        <main className={classnames({ 'active': this.state.isOpen })}>
          {this.props.children}
        </main>
      </div>
    );
  }
}
