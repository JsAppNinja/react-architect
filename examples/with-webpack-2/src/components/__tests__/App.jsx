/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

jest.mock('react-md/lib/Inks/InkContainer');

describe('App', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
