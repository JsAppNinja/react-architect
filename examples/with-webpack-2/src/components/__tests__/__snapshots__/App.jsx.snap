// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className={undefined}
  style={undefined}
>
  <header
    className="md-paper md-paper--2 md-toolbar md-background--primary md-toolbar--text-white md-toolbar--discrete md-toolbar--fixed"
    style={undefined}
  >
    <button
      className="md-inline-block md-btn md-btn--icon md-pointer--hover md-btn--toolbar md-toolbar--action-left"
      disabled={false}
      href={undefined}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <i
        className="md-icon material-icons"
        style={undefined}
      >
        menu
      </i>
    </button>
    <h2
      className="md-title md-title--toolbar"
      style={undefined}
    >
      Welcome to react-md
    </h2>
  </header>
  <main
    className="md-navigation-drawer-content md-navigation-drawer-content--inactive md-transition--acceleration md-toolbar-relative"
    id="main-content"
    style={undefined}
    tabIndex={-1}
  >
    <div
      className="md-paper md-paper--1 md-card md-background--card greeting"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="md-card-title md-card-title--primary"
        style={undefined}
      >
        <h2
          className="md-card-title--title md-text md-card-title--large"
          id={undefined}
        >
          Hello from react-md
        </h2>
      </div>
      <section
        className="md-card-text"
      >
        <p>
          This is a simple app that is using the 
          <code>
            NavigationDrawer
          </code>
           component for the layout and 
          <code>
            Card
          </code>
          , 
          <code>
            CardTitle
          </code>
          , and 
          <code>
            CardText
          </code>
           for this message.
        </p>
      </section>
      <section
        className="md-card-text"
      >
        <p>
          You can edit 
          <code>
            src/components/App.jsx
          </code>
           to start making an app. Edit 
          <code>
            src/_globals.scss
          </code>
           for any Sass variable changes.
        </p>
      </section>
    </div>
  </main>
</div>
`;
